### ISSUE TITLE
Question about parseing Opc.Ua.POWERLINK.NodeSet2.xml using open62541 XML Nodeset complier

### ISSUE THREAD

**Issue by Xiongfeng2016** [open62541/open62541#2300]
In order to understand how to parse “multiple nodesets” type XML file using open62541 XML Nodeset complier, I carefully read the last chapter of open62541 documentation.

Based on how to parse “Opc.Ua.Plc.NodeSet2.xml” introduced in last chapter of open62541 documentation, I tried to parse Opc.Ua.POWERLINK.NodeSet2.xml, which is included in github/open62541/deps/nodeset/POWERLINK folder.

I think The Powerlink nodeset “Opc.Ua.Plc.NodeSet2.xml” depends on the DI nodeset “Opc.Ua.Di.NodeSet2.xml” which then depends on NS0.
Hence, I modify the CMAKELists.txt file, of which the old version is included in open62541 source code: open62541/examples/nodeset folder. 
The modified CMAKEList.txt file is as follows:

```
# Powerlink requires the full ns0 as basis
if(UA_NAMESPACE_ZERO STREQUAL "FULL")

    # Generate types and namespace for DI
    ua_generate_nodeset_and_datatypes(
        NAME "di"
        FILE_CSV "${PROJECT_SOURCE_DIR}/deps/ua-nodeset/DI/OpcUaDiModel.csv"
        FILE_BSD "${PROJECT_SOURCE_DIR}/deps/ua-nodeset/DI/Opc.Ua.Di.Types.bsd"
        NAMESPACE_IDX 2
        FILE_NS "${PROJECT_SOURCE_DIR}/deps/ua-nodeset/DI/Opc.Ua.Di.NodeSet2.xml"
        INTERNAL
    )

    # generate powerlink namespace which is using DI
    ua_generate_nodeset_and_datatypes(
        NAME "powerlink"
        # powerlink does not define custom types. Only generate the nodeset
        FILE_NS "${PROJECT_SOURCE_DIR}/deps/ua-nodeset/POWERLINK/Opc.Ua.POWERLINK.NodeSet2.xml"
        # powerlink depends on the di nodeset, which must be generated before
        DEPENDS "di"
        INTERNAL
    )**

    add_example(server_nodeset_plcopen server_nodeset_plcopen.c
                ${PROJECT_BINARY_DIR}/src_generated/ua_types_di_generated.c
                ${PROJECT_BINARY_DIR}/src_generated/ua_namespace_di.c
                ${PROJECT_BINARY_DIR}/src_generated/ua_namespace_powerlink.c)
    add_dependencies(server_nodeset_plcopen open62541-generator-ns-powerlink)
   target_include_directories(server_nodeset_plcopen PRIVATE ${PROJECT_SOURCE_DIR}/src) # needs an internal header
endif()
```

Then, I finished the following operations as follows:
	jump to “open62541” file folder
	establish a “build” file folder, “mkdir build”
	“cd build”
	“cmake ..”
	“make”
Unfortunately, when I cloud not get the right result, I already post the wrong result as follows:
![image](https://user-images.githubusercontent.com/45161114/50227639-8b3ffa80-03e9-11e9-8432-da79cac89531.png)

If someone can reply this question, i will appreciate it. Thank you very much!!!

**Comment by Pro** [comment]
Without further information what you mean by "right result", we can not help you...

Please try to extend your issue information and also stick to the template we provide.

Did you already check our documentation?
https://open62541.org/doc/current/nodeset_compiler.html
Read all of it, especially the last chapter.

**Comment by Xiongfeng2016** [comment]
> Without further information what you mean by "right result", we can not help you...
> 
> Please try to extend your issue information and also stick to the template we provide.
> 
> Did you already check our documentation?
> https://open62541.org/doc/current/nodeset_compiler.html
> Read all of it, especially the last chapter.

Thank you for your reply so much. I am so sorry about that i don't discribe my question in detail.  Based on your suggestion, i update the description of my problem more detailed. Please check it in your convenience~~

**Comment by jpfr** [comment]
Can you confirm that #2385 fixed the issue?

**Comment by matkonnerth** [comment]
Hi, here's an example generating and building the powerlink nodeset based on master. The powerlink nodeset defines some datatypes, I've added the powerlinkTypes.bsd

**Comment by Pro** [comment]
This is fixed with #2404

I tested the example from @matkonnerth (which is now also included in the PR) and it works with that one.

Thanks for the issue report and sorry for the long fixing time.
